cunumeric/module.py

```diff
@@ -209,44 +205,30 @@ def choose(a, choices, out=None, mode="raise"):
     return array.choose(choices=choices, out=out, mode=mode, stacklevel=2)
 
 
+def diag_helper(a, offset=0, axes=None, extract=True):
+    array = ndarray.convert_to_cunumeric_ndarray(a)
+    return array.diag_helper(offset=offset, axes=axes, extract=extract)
+
+
 @copy_docstring(np.diag)
 def diag(v, k=0):
     array = ndarray.convert_to_cunumeric_ndarray(v)
     if array._thunk.scalar:
         return array.copy()
```

Comment:
Let's remove this chunk, it's almost certainly an optimization that is no longer valid. There is no case where this function can return a direct copy of the original array; even if it's a singleton array we still need to add or remove a dimension.

Reply:
Sure. I will add the error message if anyone is trying to pass scalar to the `diag` as it is done in `numpy`